engines/cine/main_loop.cpp

```diff
@@ -75,149 +75,148 @@ static void processEvent(Common::Event &event) {
 			break;
-		case Common::KEYCODE_KP0:
+		case kActionGameSpeedDefault:
 			g_cine->setDefaultGameSpeed();
 			break;
-		case Common::KEYCODE_MINUS:
-		case Common::KEYCODE_KP_MINUS:
+		case kActionGameSpeedSlower:
 			g_cine->modifyGameSpeed(-1); // Slower
 			break;
-		case Common::KEYCODE_PLUS:
-		case Common::KEYCODE_KP_PLUS:
+		case kActionGameSpeedFaster:
 			g_cine->modifyGameSpeed(+1); // Faster
 			break;
-		case Common::KEYCODE_KP4:
+		case kActionMoveLeft:
 			moveUsingKeyboard(-1, 0); // Left
 			break;
-		case Common::KEYCODE_KP6:
+		case kActionMoveRight:
 			moveUsingKeyboard(+1, 0); // Right
 			break;
-		case Common::KEYCODE_KP8:
+		case kActionMoveUp:
 			moveUsingKeyboard(0, +1); // Up
 			break;
-		case Common::KEYCODE_KP2:
+		case kActionMoveDown:
 			moveUsingKeyboard(0, -1); // Down
 			break;
-		case Common::KEYCODE_KP9:
+		case kActionMoveUpRight:
 			moveUsingKeyboard(+1, +1); // Up & Right
 			break;
-		case Common::KEYCODE_KP7:
+		case kActionMoveUpLeft:
 			moveUsingKeyboard(-1, +1); // Up & Left
 			break;
-		case Common::KEYCODE_KP1:
+		case kActionMoveDownLeft:
 			moveUsingKeyboard(-1, -1); // Down & Left
 			break;
-		case Common::KEYCODE_KP3:
+		case kActionMoveDownRight:
 			moveUsingKeyboard(+1, -1); // Down & Right
 			break;
-		case Common::KEYCODE_LEFT:
-			// fall through
-		case Common::KEYCODE_RIGHT:
-			// fall through
-		case Common::KEYCODE_UP:
-			// fall through
-		case Common::KEYCODE_DOWN:
-			// fall through
 		default:
-			g_cine->_keyInputList.push_back(event.kbd);
 			break;
-		}
+		};
 		break;
-	case Common::EVENT_KEYUP:
-		switch (event.kbd.keycode) {
-		case Common::KEYCODE_RETURN:
-		case Common::KEYCODE_KP_ENTER:
-		case Common::KEYCODE_KP5:
+	case Common::EVENT_CUSTOM_ENGINE_ACTION_END:
+		switch (event.customType) {
+		case kActionMouseLeft:
 			if (allowPlayerInput) {
 				mouseLeft = 0;
-			}
-			break;
-		case Common::KEYCODE_ESCAPE:
+			} break;
+		case kActionMouseRight:
 			if (allowPlayerInput) {
 				mouseRight = 0;
 			}
+		case kActionMoveUp:
+		case kActionMoveDown:
+		case kActionMoveLeft:
+		case kActionMoveRight:
+		case kActionMoveUpLeft:
+		case kActionMoveUpRight:
+		case kActionMoveDownLeft:
+		case kActionMoveDownRight:
+			// Stop ego movement made with keyboard when releasing a known key
+			moveUsingKeyboard(0, 0);
 			break;
-		case Common::KEYCODE_F11:
+		case kActionCollisionPage:
 			renderer->showCollisionPage(false);
 			break;
-		case Common::KEYCODE_KP4:   // Left
-		case Common::KEYCODE_KP6:   // Right
-		case Common::KEYCODE_KP8:   // Up
-		case Common::KEYCODE_KP2:   // Down
-		case Common::KEYCODE_KP9:   // Up & Right
-		case Common::KEYCODE_KP7:   // Up & Left
-		case Common::KEYCODE_KP1:   // Down & Left
-		case Common::KEYCODE_KP3:   // Down & Right
-			// Stop ego movement made with keyboard when releasing a known key
-			moveUsingKeyboard(0, 0);
+		default:
 			break;
+		}
+		break;
+	case Common::EVENT_KEYDOWN:
+		switch (event.kbd.keycode) {
```

Comment:
Seems we would not need a switch when everything falls through to the default case.
It seems the original intent was to clearly indicate these are handled like any other key, but I'm not sure it would still be clear.

Reply:
That makes sense, Fixed.